src/scenes/gamescreen.ts

```diff
@@ -61,4 +86,41 @@ export class GameScreen extends Phaser.Scene {
         }
         this.dragStart = activePointer.position.clone();
     }
+
+    /**
+     * All tiles are kept in a placedTilesContainer.\
+     * To update display, first remove all existing images, then create new ones using placed tiles from Tilemanager\
+     *
+     * Whenever information about tiles is updated, it calls a list of placed tiles from TileManager
+     *
+     * Add the image at coordinate * pixels
+     *
+     * @returns void
+     */
+    private displayPlacedTiles(): void {
+        assert(this.placedTilesContainer);
+        this.placedTilesContainer.removeAll(true);
+
+        const placedTiles: Nullable<Tile[]> = TileManager.getPlaced();
+        if (!placedTiles) {
+            return;
+        }
+        assert(placedTiles);
+        // this arrow function is only called in for loop, so should share same scope as displayedPlacedTiles()
+        placedTiles.forEach((tile) => {
+            assert(
+                tile.coordinateX &&
+                    tile.coordinateY &&
+                    this.placedTilesContainer,
+            );
```

Comment:
i think this.placedTilesContainer already gets asserted at the beginning of the method above?

Reply:
VS Code complains if i remove assert this.placedTilesContainer, probably because it is inside the arrow function?

Reply:
yess you absolutely right, i apologise. 
You can leave it like this now which would be fine... if you want you could refactor it to only use 2 lines not 5 and look a bit better: IMO prettier has some ugly ways of formatting including the current (breaking the assert into 5 lines). What i usually do then is to make it 2 seperate asserts like this: 
```
assert(tile.coordinateX && tile.coordinateY);
assert(this.placedTilesContainer);
```